modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_base_impl.ftl

```diff
@@ -243,6 +243,100 @@ import org.osgi.service.component.annotations.Reference;
 			return ${entity.variableName}Persistence.update(${entity.variableName});
 		}
 
+		<#if entity.hasExternalReferenceCode() && entity.hasLazyReference() && entity.hasEntityColumn("companyId") && stringUtil.equals(entity.name, "User")>
+			/**
+			 * Adds the incomplete ${entity.humanName} to the database. Also notifies the appropriate model listeners.
```

Comment:
point to think about: is it correct that the model listeners run on an incomplete row? Because I think that the model listener presume that all the "mandatory" attributes are there and in this case is not

Reply:
and this could be a problem because if we don't run the afterAdd on this we should run it when we put it as completed (also if it's an update)

Reply:
as discussed the listeners should take care of this scenario checking if the previous importBatchStatus is 0 or not